subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/DefaultLenientConfiguration.java

```diff
@@ -123,8 +127,11 @@ public void collectBuildDependencies(BuildDependenciesVisitor visitor) {
             @Override
             public void visitArtifacts(ArtifactVisitor visitor, boolean continueOnSelectionFailure) {
                 if (!unresolvedDependencies.isEmpty()) {
-                    for (UnresolvedDependency unresolvedDependency : unresolvedDependencies) {
-                        visitor.visitFailure(unresolvedDependency.getProblem());
+                    for (final UnresolvedDependency unresolvedDependency : unresolvedDependencies) {
+                        Throwable problem = unresolvedDependency.getProblem();
+                        visitor.visitFailure(problem);
+                        ModuleVersionSelector versionSelector = unresolvedDependency.getSelector();
+                        visitor.visitResolutionFailure(AbstractResolutionFailure.of(DefaultModuleComponentIdentifier.newId(versionSelector.getGroup(), versionSelector.getName(), versionSelector.getVersion()), problem));
```

Comment:
Coming back to my point of "why a ResolutionFailure interface?": What I'd like to suggest is instead creating public exception types that contain the relevant information:

```
Throwable rootProblem = unresolvedDependency.getProblem();
ModuleVersionSelector versionSelector = unresolvedDependency.getSelector();
ComponentIdentifier componentId = DefaultModuleComponentIdentifier.newId(versionSelector.getGroup(), versionSelector.getName(), versionSelector.getVersion());
visitor.visitFailure(new ArtifactResolutionException(componentId, problem));
```

where `ArtifactResolutionException` has a `getComponentId`, but does not have a `getAttemptedLocations`, because that would be null.